pkg/shell/cockpit-accounts.js

```diff
@@ -308,62 +363,58 @@ PageAccountsCreate.prototype = {
-        }
-
-        function highlight_error() {
-            $("#accounts-create-dialog .check-passwords").addClass("has-error");
-            $('#accounts-create-message-password-mismatch').css("visibility", "visible");
+    validate: function() {
+        var ex, fails = [];
+        var pw = $('#accounts-create-pw1').val();
+        if ($('#accounts-create-pw2').val() != pw) {
+            ex = new Error(_("The passwords do not match"));
+            ex.target = "#accounts-create-pw2";
+            fails.push(ex);
         }
-
-        function hide_error() {
-            $("#accounts-create-dialog .check-passwords").removeClass("has-error");
-            $('#accounts-create-message-password-mismatch').css("visibility", "hidden");
+        if (!$('#accounts-create-user-name').val()) {
+            ex = new Error(_("No user name specified"));
+            ex.target = '#accounts-create-user-name';
+            fails.push(ex);
         }
-
-        function check_password_match() {
-            if ($('#accounts-create-pw2').val() !== "" &&
-                $('#accounts-create-pw1').val() !== $('#accounts-create-pw2').val())
-                highlight_error();
-            else
-                hide_error();
+        if (!$('#accounts-create-real-name').val()) {
+            ex = new Error(_("No real name specified"));
+            ex.target = '#accounts-create-real-name';
+            fails.push(ex);
         }
 
-        window.clearTimeout(this.error_timeout);
-        this.error_timeout = null;
-
-        if (behavior == "changeFocus") {
-            if ($('#accounts-create-pw2').val() !== "" &&
-                $('#accounts-create-pw1').val() !== $('#accounts-create-pw2').val())
-                highlight_error();
-            else
-                hide_error();
-        } else if (behavior == "input") {
-            if ($('#accounts-create-pw2').val() !== "" &&
-                $('#accounts-create-pw1').val().indexOf($('#accounts-create-pw2').val()) !== 0)
-                highlight_error();
-            else
-                hide_error();
+        /* The first check is immediately complete */
+        var dfd = $.Deferred();
+        if (fails.length)
+            dfd.reject(fails);
+        else
+            dfd.resolve();
 
-            this.error_timeout = window.setTimeout(check_password_match, 2000);
-            this.setTimeout = null;
-        }
+        var promise = password_quality(pw)
+            .fail(function(ex) {
+                ex.target = "#accounts-create-pw2";
+            })
+            .always(function(arg) {
+                var strength = this.state() == "resolved" ? arg : "weak";
+                var meter = $("#accounts-create-password-meter")
+                    .removeClass("weak okay good excellent");
+                if (pw)
+                    meter.addClass(strength);
+                var message = $("#accounts-create-password-meter-message");
```

Comment:
I traced through the code, and `#accounts-create-password-meter-message` is never shown, even for an excellent password.